pkg/component/controller/clusterconfig.go

```diff
@@ -204,36 +203,39 @@ func (i *ClusterConfigInitializer) ensureClusterConfigExistence(ctx context.Cont
 				err = i.createClusterConfig(ctx, client)
 				if err == nil {
 					i.log.Debug("Cluster configuration created")
-					return true, nil
+					return nil
 				}
 				if apierrors.IsAlreadyExists(err) {
 					// An already existing configuration is just fine.
 					i.log.Debug("Cluster configuration already exists")
-					return true, nil
+					return nil
 				}
 			}
 		} else {
 			err = i.clusterConfigExists(ctx, client)
 			if err == nil {
 				i.log.Debug("Cluster configuration exists")
-				return true, nil
+				return nil
 			}
 		}
 
-		i.log.WithError(err).Debug("Failed to ensure the existence of the cluster configuration")
-		return false, nil
-	})
-
-	if pollErr != nil {
-		pollErr = cmp.Or(context.Cause(ctx), pollErr)
-		if err != nil {
-			return fmt.Errorf("%w (%w)", pollErr, err)
+		var log func(...any)
```

Comment:
Maybe we should consider exiting with an error at some point.
Knowing leader elector takes at least 60 seconds to deal with a stale leader, maybe it makes sense to actually crash after 5 minutes.

Reply:
Every value we put here seems kinda arbitrary. In the past years, I've hit so many cases of arbitrary timeouts breaking stuff that I I'm genuinely convinced that if unsure which timeout to pick, it's better to pick none at all (given that your caller can pick one). What I mean by that is that it's better to force users to hit Ctrl+C, or issue a `systemctl restart ...` if they think something takes too long than hard-coding a non-overridable value that will break stuff even when users know that it'll succeed eventually.